R-package/configure.win

```diff
@@ -6,7 +6,9 @@
 ###########################
 
 R_EXE="${R_HOME}/bin${R_ARCH_BIN}/R"
-CC=`"${R_EXE}" CMD config CC`
+CXX=`"${R_EXE}" CMD config CXX`
```

Comment:
Why do we use `CXX11` for Unix and `CXX` here to configure `CXX` variable (considering #4506 as well)??
https://github.com/microsoft/LightGBM/blob/b43b6c71d6b01e738b71aaad44aaa759f026ee7c/R-package/configure.ac#L24

Reply:
ah! thanks for noticing, that is an oversight. It should be CXX11 since we explicitly set a C++ standard

https://github.com/microsoft/LightGBM/blob/926526c838196a7497a85b6b8cf07657a88b69e6/R-package/src/Makevars.win.in#L1

From https://cran.r-project.org/doc/manuals/R-exts.html#Configure-and-cleanup

>  If the package specifies a non-default C++ standard, use the config variable names (such as `CXX17`) appropriate to the standard

From R's perspective, our use of C++ is "non-default". See https://cran.r-project.org/doc/manuals/R-exts.html#Using-C_002b_002b-code.

> As from R 4.0.0 a C++ compiler will be selected only if it conforms to the 2011 standard (‘C++11’). A minor update43 (‘C++14’) was published in December 2014 and will be used by default as from R 4.1.0 if supported

updated in https://github.com/microsoft/LightGBM/pull/4504/commits/b576a4f580f6b16f75a77462fb688c5df1b942ca

Reply:
Seems we are setting less variables than it's done in the example from https://cran.r-project.org/doc/manuals/R-exts.html#Using-C_002b_002b-code.
```
CXX11=`"${R_EXE}" CMD config CXX11`
CXX11STD=`"${R_EXE}" CMD config CXX11STD`
CXX="${CXX11} ${CXX11STD}"
CXXFLAGS=`"${R_EXE}" CMD config CXX11FLAGS`
CPPFLAGS=`"${R_EXE}" CMD config CPPFLAGS`
```

Also note that `CXX11FLAGS` is used to configure `CXXFLAGS`.

Reply:
I was able to find a similar code in the wild by searching for `CXX11FLAGS filename:configure.ac`. For example,
https://github.com/cbg-ethz/MC-CBN/blob/6cd55b8289d22809a5029981f87cd1cbe5be27ea/configure.ac#L21-L38
https://github.com/cran/spp/blob/2603673566115871405f5842e489454805b9f5c0/configure.ac#L15-L22
https://github.com/akhikolla/ClusterTests/blob/1c7702ba4035511ffe7d4dd27e9021b6962718fe/BART/configure.ac#L19-L22

Reply:
oooo interesting, I missed that! Thanks for pointing it out. I just made those changes here and on #4506 .